packages/material-ui-lab/src/MasonryItem/MasonryItem.js

```diff
@@ -0,0 +1,158 @@
+  getValue,
+  handleBreakpoints,
+  unstable_resolveBreakpointValues as resolveBreakpointValues,
+} from '@material-ui/system';
+import { unstable_useForkRef as useForkRef } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { getMasonryItemUtilityClass } from './masonryItemClasses';
+import MasonryContext from '../Masonry/MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryItemUtilityClass, classes);
+};
+
+export const style = ({ styleProps, theme }) => {
+  let styles = {
+    width: '100%',
+    '& > *': {
+      // all contents should have a width of 100%
+      width: '100%',
+    },
+    visibility: styleProps.height ? 'visible' : 'hidden',
+    gridColumnEnd: `span ${styleProps.columnSpan}`,
+  };
+
+  const base = Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
+    if (styleProps.spacing[breakpoint] != null) {
+      acc[breakpoint] = true;
+    }
+    return acc;
+  }, {});
+  const spacingValues = resolveBreakpointValues({ values: styleProps.spacing, base });
+  const transformer = createUnarySpacing(theme);
+  const styleFromPropValue = (propValue) => {
+    const gap = styleProps.height ? Number(getValue(transformer, propValue).replace('px', '')) : 0;
+    const rowSpan = styleProps.height ? Math.ceil(styleProps.height + gap) : 0;
+    return {
+      gridRowEnd: `span ${rowSpan}`,
+      paddingBottom: gap - 1,
+    };
+  };
+  styles = { ...styles, ...handleBreakpoints({ theme }, spacingValues, styleFromPropValue) };
+
+  return styles;
+};
+
+const MasonryItemRoot = styled('div', {
+  name: 'MuiMasonryItem',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    return [styles.root];
+  },
+})(style);
+
+const MasonryItem = React.forwardRef(function MasonryItem(inProps, ref) {
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiMasonryItem',
+  });
+
+  const masonryItemRef = React.useRef(null);
+
+  const { spacing = 1 } = React.useContext(MasonryContext);
+  const { children, className, component = 'div', columnSpan = 1, height, ...other } = props;
+  const [styleProps, setStyleProps] = React.useState({
+    ...props,
+    spacing,
+    columnSpan,
+    height: height < 0 ? 0 : height, // MasonryItems to which negative or zero height is passed will be hidden
+  });
+
+  const classes = useUtilityClasses(styleProps);
+
+  React.useEffect(() => {
+    const resizeObserver =
+      height === undefined
+        ? new ResizeObserver(([item]) => {
+            setStyleProps({
+              ...styleProps,
+              height: item.contentRect.height,
+            });
+          })
+        : null; // If height is passed by user, ResizeObserver is not used.
+    if (resizeObserver && masonryItemRef?.current) {
```

Comment:
```suggestion
    if (resizeObserver && masonryItemRef.current) {
```
`masonryItemRef` is always defined.

What should happen if the instance isn't resolved yet?